test/reaping-faulty-nodes.js

```diff
@@ -121,3 +127,107 @@ test2('tombstone should not be applied when gossiped about but unknown to the SU
         ];
     })
 );
+
+test2('test /admin/reap endpoint', getClusterSizes(2), 20000,
+    prepareWithStatus(1, 'faulty', function(t, tc, n) { return [
+        dsl.assertStats(t, tc, {alive: n, faulty: 1}, {1: {status: 'faulty'}}),
+
+        dsl.callEndpoint(t, tc, '/admin/reap'),
+        dsl.validateEventBody(t, tc, {
+            type: events.Types.AdminReap,
+            direction: 'response'
+        }, 'Wait for AdminReap response', function (response) {
+            return response.body.status === 'ok';
+        }),
+
+        dsl.assertStats(t, tc, {alive: n, tombstone: 1}, {1: {status: 'tombstone'}}),
+    ];})
+);
+
+test2('tombstone should be applied when sent as a state', getClusterSizes(2), 20000,
+    prepareCluster(function(t, tc, n) {
+        return [
+            dsl.sendPing(t, tc, 0, {
+                sourceIx: 0,
+                subjectIx: 1,
+                status: 'tombstone'
+            }),
+            dsl.waitForPingResponse(t, tc, 0),
+
+            // confirm that the tombstone has not been added to the membership list
+            dsl.assertStats(t, tc, {
+                alive: n,
+                tombstone: 1,
+            }),
+        ];
+    })
+);
+
+test2('tombstone should be gossiped with flag when applied as state', getClusterSizes(2), 20000,
+    prepareCluster(function(t, tc, n) {
+        return [
+            dsl.sendPing(t, tc, 0, {
+                sourceIx: 0,
+                subjectIx: 1,
+                status: 'tombstone'
+            }),
+            dsl.waitForPingResponse(t, tc, 0),
+
+            // clear all pings that happened before we gossiped the tombstone
+            dsl.consumePings(t, tc),
+
+            // Wait for a ping from the SUT and validate that it does not gossip about the tombstone
+            dsl.validateEventBody(t, tc, {
+                type: events.Types.Ping,
+                direction: 'request'
+            }, "The gossip should contain a flagged tombstone", function (ping) {
+                return _.filter(ping.body.changes, { status: 'faulty', tombstone: true }).length === 1
+                    && _.filter(ping.body.changes, { status: 'tombstone' }).length === 0;
+            }),
+        ];
+    })
+);
+
+test2('tombstone should be applied when sent as a flag', getClusterSizes(2), 20000,
+    prepareCluster(function(t, tc, n) {
+        return [
+            dsl.sendPing(t, tc, 0, {
+                sourceIx: 0,
+                subjectIx: 1,
+                status: 'tombstoneFlag'
```

Comment:
Is it possible to somehow just set the `tombstone: true` flag here? It took me a minute to figure out what was going on because 'tombstoneFlag' is not a real status...